Boost not working with FetchContent_MakeAvailable(SetupHunter)

I have been using `FetchContent_MakeAvailable(SetupHunter)` with multiple different packages successfully already, but now I tried it with Boost and it failed with the following errors:
```
CMake Error at C:/.hunter/_Base/b67e11d/083dd95/9877faf/Install/lib/cmake/Boost/BoostConfig.cmake:4 (include):
  include could not find load file:

    hunter_check_toolchain_definition
Call Stack (most recent call first):
  CMakeLists.txt:15 (find_package)


CMake Error at C:/.hunter/_Base/b67e11d/083dd95/9877faf/Install/lib/cmake/Boost/BoostConfig.cmake:7 (include):
  include could not find load file:

    hunter_internal_error
Call Stack (most recent call first):
  CMakeLists.txt:15 (find_package)


CMake Error at C:/.hunter/_Base/b67e11d/083dd95/9877faf/Install/lib/cmake/Boost/BoostConfig.cmake:8 (include):
  include could not find load file:

    hunter_status_debug
Call Stack (most recent call first):
  CMakeLists.txt:15 (find_package)



[hunter ** INTERNAL **] Unexpected empty string
[hunter ** INTERNAL **] [Directory:C:/.hunter/_Base/b67e11d/083dd95/9877faf/Install/lib/cmake/Boost]

------------------------------ ERROR -----------------------------
    /error.internal.html
------------------------------------------------------------------

CMake Error at C:/.hunter/_Base/Download/Hunter/0.23.234/b67e11d/Unpacked/cmake/modules/hunter_error_page.cmake:12 (message):
Call Stack (most recent call first):
  C:/.hunter/_Base/Download/Hunter/0.23.234/b67e11d/Unpacked/cmake/modules/hunter_internal_error.cmake:13 (hunter_error_page)
  C:/.hunter/_Base/Download/Hunter/0.23.234/b67e11d/Unpacked/cmake/modules/hunter_assert_not_empty_string.cmake:9 (hunter_internal_error)
  C:/.hunter/_Base/Download/Hunter/0.23.234/b67e11d/Unpacked/cmake/modules/hunter_check_toolchain_definition.cmake:46 (hunter_assert_not_empty_string)
  C:/.hunter/_Base/b67e11d/083dd95/9877faf/Install/lib/cmake/Boost/BoostConfig.cmake:43 (hunter_check_toolchain_definition)
  CMakeLists.txt:15 (find_package)
```
The CMakeLists.txt looks like follows:
```cmake
cmake_minimum_required(VERSION 3.2)

set(HUNTER_Boost_COMPONENTS chrono date_time)
set(HUNTER_PACKAGES Boost)
set(HUNTER_Boost_VERSION 1.71.0-p0)

# option(HUNTER_NO_TOOLCHAIN_ID_RECALCULATION "" ON)

include(FetchContent)
FetchContent_Declare(SetupHunter GIT_REPOSITORY https://github.com/cpp-pm/gate)
FetchContent_MakeAvailable(SetupHunter)

project(use-pkg-components)

find_package(Boost CONFIG REQUIRED chrono date_time)

add_executable(use-pkg-components main.cpp)

target_link_libraries(use-pkg-components PRIVATE
    Boost::chrono
    Boost::date_time
)
```

Interestingly the same basic setup works fine when I manually download & set up HunterGate.cmake:
```cmake
cmake_minimum_required(VERSION 3.2)

set(HUNTER_Boost_COMPONENTS chrono date_time)
set(HUNTER_PACKAGES Boost)
set(HUNTER_Boost_VERSION 1.71.0-p0)

# option(HUNTER_NO_TOOLCHAIN_ID_RECALCULATION "" ON)

include("cmake/HunterGate.cmake")
HunterGate(
    URL "https://github.com/cpp-pm/hunter/archive/v0.23.234.tar.gz"
    SHA1 "b67e11d7e9ee503e9f6e1b45eb1ef7bf2fb31279"
)

project(use-pkg-components)

hunter_add_package(Boost COMPONENTS chrono date_time)
find_package(Boost CONFIG REQUIRED chrono date_time)

add_executable(use-pkg-components main.cpp)

target_link_libraries(use-pkg-components PRIVATE
    Boost::chrono
    Boost::date_time
)
```

Any ideas what could be the issue ?
Thanks

I've tried your snippet and got a different error with MinGW 8.1.0

```
[6/8 0.0/sec]Performing build step for 'Boost-chrono'
FAILED: Boost-chrono-prefix/src/Boost-chrono-stamp/Boost-chrono-build
cmd.exe /C "cd /D C:\.hunter\_Base\d8992c8\a76317a\d6df02a\Build\Boost\__chrono\Source && b2 target-os=windows -a link=static threading=multi variant=debug,release --layout=tagged toolset=gcc --user-config=C:/.hunter/_Base/d8992c8/a76317a/d6df02a/Build/Boost/__chrono/Build/boost.user.jam --with-chrono -j 8 --ignore-site-config && C:\Tools\cmake\bin\cmake.exe -E touch C:/.hunter/_Base/d8992c8/a76317a/d6df02a/Build/Boost/__chrono/Build/Boost-chrono-prefix/src/Boost-chrono-stamp/Boost-chrono-build"
error: at project-config.jam:6
error: duplicate initialization of gcc with the following parameters:
error: version = 8.1.0
error: previous initialization at C:\.hunter\_Base\d8992c8\a76317a\d6df02a\Build\Boost\__chrono\Build\boost.user.jam:6

ninja: build stopped: subcommand failed.

[hunter ** FATAL ERROR **] Build step failed (dir: C:/.hunter/_Base/d8992c8/a76317a/d6df02a/Build/Boost/__chrono
[hunter ** FATAL ERROR **] [Directory:C:/.hunter/_Base/Download/Hunter/0.23.235/d8992c8/Unpacked/cmake/projects/Boost/chrono]

------------------------------ ERROR -----------------------------
    https://cpp-pm-hunter.readthedocs.io/en/latest/reference/errors/error.external.build.failed.html
------------------------------------------------------------------

CMake Error at C:/.hunter/_Base/Download/Hunter/0.23.235/d8992c8/Unpacked/cmake/modules/hunter_error_page.cmake:12 (message):
Call Stack (most recent call first):
  C:/.hunter/_Base/Download/Hunter/0.23.235/d8992c8/Unpacked/cmake/modules/hunter_fatal_error.cmake:20 (hunter_error_page)
  C:/.hunter/_Base/Download/Hunter/0.23.235/d8992c8/Unpacked/cmake/modules/hunter_download.cmake:617 (hunter_fatal_error)
  C:/.hunter/_Base/Download/Hunter/0.23.235/d8992c8/Unpacked/cmake/projects/Boost/chrono/hunter.cmake:16 (hunter_download)
  C:/.hunter/_Base/Download/Hunter/0.23.235/d8992c8/Unpacked/cmake/modules/hunter_add_package.cmake:62 (include)
  build/_deps/setuphunter-src/CMakeLists.txt:142 (hunter_add_package)


-- Configuring incomplete, errors occurred!
```

The only difference between manual including the `HunterGate` and uising `FetchContent` would be the latter using `add_subdirectory`. 

I don't know if this is problem with `boost` or with packages that have `COMPONENTS`.

The exact error that I am currently getting is:
```
-- *** *** *** CMAKE_MODULE_PATH = "/home/carlo/projects/aicxx/ai-statefultask-testsuite/ai-statefultask-testsuite/cwm4/cmake"
CMake Error at cwm4/hunter/_Base/xxxxxxx/b5a07b2/a8d2191/Install/lib/cmake/Boost/BoostConfig.cmake:6 (include):
  include could not find load file:

    hunter_check_toolchain_definition
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindBoost.cmake:443 (find_package)
  cwds/CMakeLists.txt:35 (find_package)
```

In the end I opted for repeating the call to `HunterGate` (with 'fake' URL and SHA1 which are ignored anyway):

```
# FetchContent_MakeAvailable is available since CMake version 3.14.
cmake_minimum_required( VERSION 3.14 )

# Use local git submodule for Hunter.
set( HUNTER_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/cwm4/hunter" ) 

# Setting up external packages.
set( HUNTER_Boost_COMPONENTS filesystem iostreams program_options system )
set( HUNTER_PACKAGES Boost farmhash )

# Initialize Hunter and build external pacakges.
include( FetchContent )
# Use local git submodule in the source tree.
FetchContent_Declare( SetupHunter SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/cwm4/hunter/gate" )
FetchContent_MakeAvailable( SetupHunter )

# Refresh CMAKE_MODULE_PATH (this call isn't using the URL or SHA1, what is really being
# used can be found in the generated ${CMAKE_CURRENT_BINARY_DIR}/HunterGate.cmake.
HunterGate(
    URL "https://github.com/cpp-pm/hunter/archive/v0.23.240.tar.gz"
    SHA1 "ca19f3769e6c80cfdd19d8b12ba5102c27b074e0"
)
```

*However* - although this uses, as intended, my git clone (fork) at `HUNTER_ROOT`, it also downloads and compiles all third-party stuff in that `${HUNTER_ROOT}/_Base` which is unacceptable. `_Base` is catenated to the hunter_root, or HUNTER_ROOT or HUNTER_CACHED_ROOT etc all over the place, so cannot be separated it seems.

Because I didn't have to patch Hunter yet I tried to comment out the setting of `HUNTER_ROOT` in the above, but that leads to the following error:

```
CMake Error at /usr/share/cmake-3.16/Modules/ExternalProject.cmake:2530 (message):
  URL_HASH is set to

    SHA1=xxxxxxxxxx

  but must be ALGO=value where ALGO is

    MD5|SHA1|SHA224|SHA256|SHA384|SHA512|SHA3_224|SHA3_256|SHA3_384|SHA3_512

  and value is a hex string.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 # Copyright (c) 2013-2014, Ruslan Baratov
-# Copyright (c) 2019 Cristian Adam
+# Copyright (c) 2019-2020 Cristian Adam
 # All rights reserved.
 
 cmake_minimum_required(VERSION 3.2)
@@ -148,8 +148,16 @@ if (ORIGINAL_TOOLCHAIN_FILE)
 endif()
 
 #
-# Make sure the parent find_package calls work as expected
+# Make sure the parent find_package and include calls work as expected
 #
 
-set(CMAKE_FIND_ROOT_PATH ${CMAKE_FIND_ROOT_PATH} PARENT_SCOPE)
-set(CMAKE_PREFIX_PATH ${CMAKE_PREFIX_PATH} PARENT_SCOPE)
+set(CMAKE_FIND_ROOT_PATH "${CMAKE_FIND_ROOT_PATH}" PARENT_SCOPE)
+set(CMAKE_PREFIX_PATH "${CMAKE_PREFIX_PATH}" PARENT_SCOPE)
+set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}" PARENT_SCOPE)
+
+#
+# Variables expected to be found in global scope
+#
+set(HUNTER_ID_PATH "${HUNTER_ID_PATH}" PARENT_SCOPE)
+set(HUNTER_TOOLCHAIN_ID_PATH "${HUNTER_TOOLCHAIN_ID_PATH}" PARENT_SCOPE)
+set(HUNTER_CONFIG_ID_PATH "${HUNTER_CONFIG_ID_PATH}" PARENT_SCOPE)

